tools/Custom/HttpMessageLogFormatter.cs

```diff
@@ -0,0 +1,107 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph.PowerShell
+{
+    using Hyak.Common;
+    using Newtonsoft.Json;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Text;
+    using System.Text.RegularExpressions;
+    using System.Xml.Linq;
+
+    public static class HttpMessageLogFormatter
+    {
+        public static string GetHttpRequestLog(HttpRequestMessage request)
+        {
+            if (request == null) return string.Empty;
+
+            string body = (request.Content == null) ? string.Empty : FormatString(request.Content.ReadAsStringAsync().Result);
+
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.AppendLine($"============================ HTTP REQUEST ============================{Environment.NewLine}");
+            stringBuilder.AppendLine($"HTTP Method:{Environment.NewLine}{request.Method.ToString()}{Environment.NewLine}");
+            stringBuilder.AppendLine($"Absolute Uri:{Environment.NewLine}{request.RequestUri.ToString()}{Environment.NewLine}");
+            stringBuilder.AppendLine($"Headers:{Environment.NewLine}{HeadersToString(ConvertHttpHeadersToCollection(request.Headers))}{Environment.NewLine}");
```

Comment:
We sanitize the headers in [`ConvertHttpHeadersToCollection`](https://github.com/microsoftgraph/msgraph-sdk-powershell/pull/868/files/b08571e6bc56d72f7a4393b61772c330a3ac2897#diff-a2f4ca8f3e28a05edec4d0643e1ec9b06eb446ef8f518dd94dc58c2687dfb4cbR64-R68) by removing `Authorization` header from the logs since it can contain PIIs when decoded.